### PR TITLE
fix(migration): return null for Image.url

### LINKED ISSUE #3322: 🐛 [Bug] Image.url returns undefined and crashes site
> [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-03-23T17:43:54Z" title="Monday, March 23rd 2020, 6:43:54 pm +01:00">Mar 23, 2020</time>_
_Closed <time datetime="2020-03-24T12:41:24Z" title="Tuesday, March 24th 2020, 1:41:24 pm +01:00">Mar 24, 2020</time>_
---

## :bug: Bugreport

Image.url should never return undefined

### Steps to reproduce the behavior
1. dump production data locally
2. visit site
3. site crashes on index
4. 
5. Profit

### PR DESCRIPTION
**Pull request by Tirokk**
> [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-03-23T17:48:13Z" title="Monday, March 23rd 2020, 6:48:13 pm +01:00">Mar 23, 2020</time>_
_Merged <time datetime="2020-03-24T12:41:24Z" title="Tuesday, March 24th 2020, 1:41:24 pm +01:00">Mar 24, 2020</time>_
---

### Issues

- fixes #3322 

should we add an id to `Image`, which is non-nullable?
can we somehow not return `deleted` users in our following/followedBy lists?

### CODE REVIEW

**Comment by Tirokk** [comment]
> [cypress[bot]](https://github.com/apps/cypress) **Authored by [cypress[bot]](https://github.com/apps/cypress)**
_<time datetime="2020-03-23T18:29:54Z" title="Monday, March 23rd 2020, 7:29:54 pm +01:00">Mar 23, 2020</time>_

---



<p align="center">
  <br />
  
</p>

---

### Test summary
<a href="https://dashboard.cypress.io/projects/qa7fe2/runs/6338/specs"></a> <b title="Passed">66</b>  &bull; <a href="https://dashboard.cypress.io/projects/qa7fe2/runs/6338/failures"></a> <b title="Failed">0</b>  &bull;  <b title="Pending">0</b> &bull;  <b title="Skipped">0</b>

---

### Run details
| | |
| --- | --- |
| **Project** | Human-Connection |
| **Status** | Passed |
| **Commit** | 6e0459ca2d |
| **Started** | Mar 23, 2020 9:06 PM |
| **Ended** | Mar 23, 2020 9:28 PM |
| **Duration** | 21:19 <a href="https://on.cypress.io/parallelization" title="Enable parallelization to reduce run duration">💡</a> |
| **OS** | Linux Ubuntu Linux - 16.04 |
| **Browser** | Firefox 68  |

[View run in Cypress Dashboard ➡️](https://dashboard.cypress.io/projects/qa7fe2/runs/6338/specs)



---

This comment has been generated by **cypress-bot** as a result of this project's GitHub integration settings. You can manage this integration in [this project's settings in the Cypress Dashboard](https://dashboard.cypress.io/projects/qa7fe2/settings)

**Comment by Tirokk** [comment]
> [cypress[bot]](https://github.com/apps/cypress) **Authored by [cypress[bot]](https://github.com/apps/cypress)**
_<time datetime="2020-03-23T18:32:24Z" title="Monday, March 23rd 2020, 7:32:24 pm +01:00">Mar 23, 2020</time>_

---



<p align="center">
  <br />
  
</p>

---

### Test summary
<a href="https://dashboard.cypress.io/projects/qa7fe2/runs/6336/specs"></a> <b title="Passed">66</b>  &bull; <a href="https://dashboard.cypress.io/projects/qa7fe2/runs/6336/failures"></a> <b title="Failed">0</b>  &bull;  <b title="Pending">0</b> &bull;  <b title="Skipped">0</b>

---

### Run details
| | |
| --- | --- |
| **Project** | Human-Connection |
| **Status** | Passed |
| **Commit** | 4e8f0076d1 <span style="cursor:help;" title="Your CI provider generated this commit by merging this branch into the target branch.">ℹ️</span> |
| **Started** | Mar 23, 2020 8:14 PM |
| **Ended** | Mar 23, 2020 8:38 PM |
| **Duration** | 23:14 <a href="https://on.cypress.io/parallelization" title="Enable parallelization to reduce run duration">💡</a> |
| **OS** | Linux Ubuntu Linux - 16.04 |
| **Browser** | Firefox 68  |

[View run in Cypress Dashboard ➡️](https://dashboard.cypress.io/projects/qa7fe2/runs/6336/specs)



---

This comment has been generated by **cypress-bot** as a result of this project's GitHub integration settings. You can manage this integration in [this project's settings in the Cypress Dashboard](https://dashboard.cypress.io/projects/qa7fe2/settings)

File: backend/src/middleware/softDelete/softDeleteMiddleware.js
Comment (Tirokk) [inline backend/src/middleware/softDelete/softDeleteMiddleware.js:20]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2020-03-24T12:38:04Z" title="Tuesday, March 24th 2020, 1:38:04 pm +01:00">Mar 24, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -17,10 +17,10 @@ const obfuscate = async (resolve, root, args, context, info) => {
     root.contentExcerpt = 'UNAVAILABLE'
     root.title = 'UNAVAILABLE'
     root.slug = 'UNAVAILABLE'
-    root.avatar = 'UNAVAILABLE'
+    root.avatar = null
```


Good catch :+1:

File: backend/src/schema/resolvers/users/location.spec.js
Comment (Tirokk) [inline backend/src/schema/resolvers/users/location.spec.js:22]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2020-03-24T12:38:29Z" title="Tuesday, March 24th 2020, 1:38:29 pm +01:00">Mar 24, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -19,48 +19,48 @@ const updateUserMutation = gql`
 const newlyCreatedNodesWithLocales = [
   {
     city: {
-      lng: -74.5763,
-      lat: 41.1534,
+      id: expect.stringContaining('place'),
```


Why do these ids change? Do we make API calls during testing?

**Comment by Tirokk** [comment]
> [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-03-24T12:47:44Z" title="Tuesday, March 24th 2020, 1:47:44 pm +01:00">Mar 24, 2020</time>_

---


I asked the question about an id that is `non-nullable` at the point directly after I had removed the `non-nullable` from the `url`, but I've since added it back because we can use the `softDeleteMiddleware` to set these values to null, if a user is deleted.

I like your idea to remove the deleted user from the following list, but I couldn't work out how to do that yesterday. I guess it could be a function that adds a filter like `filterForMutedUsers`, but `filterForDeletedUsers`?

**Review by Mogge — APPROVED** [review]
I like it! :+1:

But maybe it would be good to wait for a second approval from someone with more insight into the database...

**Review by Mogge — APPROVED** [review]
@mattwr18 to answer your questions:

> should we add an id to Image, which is non-nullable?

Why?

> can we somehow not return deleted users in our following/followedBy lists?

No, I would instead remove a deleted user from all users following collections once he/she gets deleted.

**Pull request closed without merging**
